src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java

```diff
@@ -0,0 +1,177 @@
+        return film;
+    }
+
+
+    @Override
+    public Film updateFilm(Film film) {
+        String sql = "UPDATE films SET name = ?, description = ?, release_date = ?, duration = ?, mpa_id = ? WHERE id = ?";
+        jdbcTemplate.update(sql, film.getName(), film.getDescription(), film.getReleaseDate(),
+                film.getDuration(), film.getMpa().getId(), film.getId());
+
+        jdbcTemplate.update("DELETE FROM film_genres WHERE film_id = ?", film.getId());
+        insertGenres(film);
+        film.setGenres(findGenresByFilmId(film.getId()));
+        film.setMpa(findMpaById(film.getMpa().getId()));
+        return film;
+    }
+
+    @Override
+    public List<Film> findAll() {
+        String sql = "SELECT * FROM films";
+        List<Film> films = jdbcTemplate.query(sql, filmRowMapper);
+        for (Film film : films) {
+            film.setGenres(findGenresByFilmId(film.getId()));
+            film.setMpa(findMpaById(film.getMpa().getId()));
+        }
+        return films;
+    }
+
+    @Override
+    public Film findById(int id) {
+        String sql = "SELECT * FROM films WHERE id = ?";
+        Film film = jdbcTemplate.queryForObject(sql, filmRowMapper, id);
+        film.setGenres(findGenresByFilmId(film.getId()));
+        film.setMpa(findMpaById(film.getMpa().getId()));
+        return film;
+    }
+
+    @Override
+    public void deleteAll() {
+        jdbcTemplate.update("DELETE FROM film_genres");
+        jdbcTemplate.update("DELETE FROM film_likes");
+        jdbcTemplate.update("DELETE FROM films");
+    }
+
+    // --- Gestione LIKE ---
+    @Override
+    public void addLike(int filmId, int userId) {
+        String sql = "INSERT INTO film_likes (film_id, user_id) VALUES (?, ?)";
+        jdbcTemplate.update(sql, filmId, userId);
+    }
+
+    @Override
+    public void removeLike(int filmId, int userId) {
+        String sql = "DELETE FROM film_likes WHERE film_id = ? AND user_id = ?";
+        jdbcTemplate.update(sql, filmId, userId);
+    }
+
+    @Override
+    public List<Film> getPopularFilms(int count) {
+        String sql = "SELECT f.* " +
+                "FROM films f LEFT JOIN film_likes l ON f.id = l.film_id " +
+                "GROUP BY f.id " +
+                "ORDER BY COUNT(l.user_id) DESC " +
+                "LIMIT ?";
+        List<Film> films = jdbcTemplate.query(sql, filmRowMapper, count);
+        for (Film film : films) {
+            film.setGenres(findGenresByFilmId(film.getId()));
+            film.setMpa(findMpaById(film.getMpa().getId()));
+        }
+        return films;
+    }
+
+    // --- Helpers ---
+
+    private void insertGenres(Film film) {
+        if (film.getGenres() == null || film.getGenres().isEmpty()) return;
+        String sql = "INSERT INTO film_genres (film_id, genre_id) VALUES (?, ?)";
+        for (Genre genre : film.getGenres()) {
+            jdbcTemplate.update(sql, film.getId(), genre.getId());
+        }
```

Comment:
Вместо цикла стоит рассмотреть возможность добавления пачками через batchUpdate()

(Instead of a loop, consider adding in batches via batchUpdate())